### PR TITLE
Get user endpoint

### PR DESCRIPTION
**Pull request by berkatil**
I have written the get user endpoint. Unittests should be written

### CODE REVIEW

File: back-end/sports_platform/sports_platform_api/views/user_views.py [hunk 1]
```diff
@@ -1,15 +1,36 @@
...
-import django.contrib.auth 
+from ..models import User
+from ..serializers.user_serializer import UserSerializer
+from ..validation import user_validation
+
 
+@api_view(['GET'])
+def get_user(request, user_id):
+    if not request.user.is_authenticated:
+        return Response(data={'message': 'User is not logged in, first you need to login'}, status=401)
```
Comment (fileames) [inline back-end/sports_platform/sports_platform_api/views/user_views.py:15]: Should the user be logged in to be viewed by another user? Maybe we can ask this to other teams.

File: back-end/sports_platform/sports_platform_api/views/user_views.py [hunk 2]
```diff
@@ -25,7 +46,8 @@ def create_user(request):
         guest.register(db_data)
     except ValueError:
         return Response(data = {"message": 'There is an error regarding the provided data'}, status=400)
-    except IntegrityError:
+    except IntegrityError as e:
+        print(e)
```
Comment (fileames) [inline back-end/sports_platform/sports_platform_api/views/user_views.py:50]: We can delete the print statement if not needed.

File: back-end/sports_platform/sports_platform_api/views/user_views.py [hunk 3]
```diff
@@ -1,15 +1,36 @@
...
+    except Exception:
+        return Response(data={'message': 'An error occured, please try again later.'}, status=500)
+    
+    serialized_user = UserSerializer(user).data
+    # drop last_login field that we  do not use
+    serialized_user.pop('last_login')
+    # add activity stream data
+    serialized_user['@context'] = 'https://schema.org/Person'
+    serialized_user['@id'] = user.user_id
+    serialized_user['sports'] = sports
```
Comment (fileames) [inline back-end/sports_platform/sports_platform_api/views/user_views.py:30]: For this field, we should make it consistent with the schema.

File: back-end/sports_platform/sports_platform_api/views/user_views.py (hunk 2 above)
Comment (berkatil) [reply back-end/sports_platform/sports_platform_api/views/user_views.py:50]: sure, sorry for this.

File: back-end/sports_platform/sports_platform_api/views/user_views.py (hunk 1 above)
Comment (berkatil) [reply back-end/sports_platform/sports_platform_api/views/user_views.py:15]: I do not understand what you mean. Users should **not** be logged in to be viewed by another user but they should be **logged** **in** to view other users.
I mean, for this endpoint we have 2 users:
1. User that wants to see another user. This must be logged in
2. User that will be shown. This does not have to be logged in.

Is this what you understood?

Comment (fileames) [reply back-end/sports_platform/sports_platform_api/views/user_views.py:15]: I am talking about the first case. For example I can still see the profile page of an Instagram user even when I am not logged in to Instagram. So maybe we can also provide this to not logged-in people?

File: back-end/sports_platform/sports_platform_api/views/user_views.py (hunk 3 above)
Comment (berkatil) [reply back-end/sports_platform/sports_platform_api/views/user_views.py]: I have changed it feel free to review and if it is ok, please resolve this

File: back-end/sports_platform/sports_platform_api/views/user_views.py (hunk 1 above)
Comment (berkatil) [reply back-end/sports_platform/sports_platform_api/views/user_views.py]: I have changed it such that login is not required anymore.

**Pull request merged**